test/utils/wabt_engine.cpp

```diff
@@ -0,0 +1,110 @@
+#include "src/binary-reader.h"
+#include "src/cast.h"
+#include "src/feature.h"
+#include "src/interp/binary-reader-interp.h"
+#include "src/interp/interp.h"
+#include "src/literal.h"
+#include "src/option-parser.h"
+#include "src/resolve-names.h"
+#include "src/stream.h"
+#include "src/validator.h"
+#include "src/wast-lexer.h"
+#include "src/wast-parser.h"
+
+#include <src/interp/interp.h>
+#include <test/utils/wasm_engine.hpp>
+
+namespace fizzy::test
+{
+class WabtEngine : public WasmEngine
+{
+    wabt::interp::Environment m_env;
+    wabt::interp::DefinedModule* m_module{nullptr};
+
+public:
+    void parse(bytes_view input) final;
+    std::optional<FuncRef> find_function(std::string_view name) const final;
+    void instantiate() final;
+    bytes_view get_memory() const final;
+    void set_memory(bytes_view memory) final;
+    Result execute(FuncRef func_ref, const std::vector<uint64_t>& args) final;
+};
+
+std::unique_ptr<WasmEngine> create_wabt_engine()
+{
+    return std::make_unique<WabtEngine>();
+}
+
+void WabtEngine::parse(bytes_view input)
+{
+    wabt::ReadBinaryOptions options{
+        wabt::Features{},
+        nullptr,  // debugging stream for loading
+        false,    // ReadDebugNames
+        true,     // StopOnFirstError
+        false     // FailOnCustomSectionError
+    };
+
+    wabt::Errors errors;
+    wabt::Result result =
+        wabt::ReadBinaryInterp(&m_env, input.data(), input.size(), options, &errors, &m_module);
+
+    (void)result;
+    // FIXME: parse() should return bool.
+    assert(result == wabt::Result::Ok);
+}
```

Comment:
```suggestion
    assert(m_env.GetMemoryCount() ==1);
}
```

Reply:
Should not be needed for benchmarks not using memory.

Reply:
I didn't see bench has a test for memory not being present.